importer/tasks.py

```diff
@@ -3,11 +3,55 @@
 """
 from __future__ import unicode_literals
 
+import logging
+
+from django.conf import settings
+
 from lore.celery import async
-from importer.api import import_course_from_file
+from learningresources.api import update_xanalytics
+from xanalytics import send_request, get_result, get_xanalytics_file
+
+log = logging.getLogger(__name__)
 
 
 @async.task
 def import_file(path, repo_id, user_id):
     """Asynchronously import a course."""
+    from importer.api import import_course_from_file
     import_course_from_file(path, repo_id, user_id)
+
+
+@async.task
+def populate_xanalytics_fields(course_id):
+    """
+    Initiate request to xanalytics API to get stats for a course,
+    then trigger async job to retrieve results when they become available.
+    Args:
+        course_id (int): primary key of a Course
+    """
+    if settings.XANALYTICS_URL != "":
+        token = send_request(settings.XANALYTICS_URL + "/create", course_id)
+        check_for_results.apply_async(
```

Comment:
Why not just use the request builtin mechanism for retries: i.e. use session and do `session.mount(urlbase, HTTPAdapter(max_retries=self.RETRIES))` and just handle `requests.RequestException`, it would simplify this code quite a bit I think, and should mean we can unwrap another layer of async calls.